cpp/include/proxy_sequences.h

```diff
@@ -118,7 +118,30 @@ class UpdateUploadObjectFromFile final : public Update {
   const alba::Checksum *_cs_o;
 };
 
-/* TODO upload object from in memory blob */
+class UpdateUploadObject final : public Update {
+public:
+  UpdateUploadObject(const std::string &name,
+                     const std::shared_ptr<std::vector<char>> data,
```

Comment:
yes, a pointer to the blob + size is most generic, but the copy might happen way later, so you need to keep the buffer alive until it happens.